node/middlewares/internals/utils.ts

```diff
@@ -47,3 +49,26 @@ export const getPath = async (
   }
   return path
 }
+
+export const getBindings = (
+  tenantInfo: any,
+  salesChannels: Array<Maybe<SalesChannel>> | null | undefined
+): string[] => {
+  if (!tenantInfo || !salesChannels || salesChannels.length === 0) {
+    return ['*']
+  }
+  const mapSalesChannelToBindingId = tenantInfo.bindings.reduce(
+    (acc: Record<string, string>, { id, extraContext }: any) => {
+      const salesChannelId = extraContext.portal?.salesChannel
+      if (salesChannelId) {
+        acc[salesChannelId] = id
+      }
+      return acc
+    },
+    {} as Record<string, string>
+  )
+
+  return salesChannels.map(
```

Comment:
I think it might be possible to have tenant and have no binding?
If so, mapSalesChannelToBindingId would be empty and we would have a map with [nulls].

Reply:
You are correct, I changed it to a reduce that removes nulls